### ISSUE TITLE
Screenshot of non Steam games can't be uploaded

### ISSUE THREAD

**Issue by strycore** [ValveSoftware/SteamOS#779]
#### Your system information

* Steam client version: N/A
* SteamOS version: 3.1
* Opted into Steam client beta?: N/A
* Opted into SteamOS beta?: No
* Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:

When taking a screenshot of a non Steam game and trying to upload it, a notification appears saying "Failed to upload screenshot"
Screenshots of Steam games or of the SteamOS UI upload without issue.

#### Steps for reproducing this issue:

1. Add a non Steam game to the Deck
2. Take a screenshot with Steam + R1
3. Try uploading it to Steam
4. Message "Failed to upload screenshot" appears

**Comment by TTimo** [comment]
Hello,

Since the game doesn't exist on the Steam servers, so there is nowhere to upload and store screenshots for it.

**Comment by strycore** [comment]
I figured it was something like that but on the other hand it's perfectly possible to upload screenshots of the Deck UI to Steam which isn't tied to any game at all. I would have expected screenshots of non-Steam games to be considered as non game screenshots as well.

**Comment by TTimo** [comment]
You're right. Screenshots are tied to your Steam account, so maybe it'd make sense if it could take screenshot uploads from non-steam games. The origin shouldn't really matter.

**Comment by NightHammer1000** [comment]
Maybe a Option to Upload these kind of Screenshots somewhere else like [imgur?](https://imgur.com/)

**Comment by arrowgent** [comment]
you can put them in your "artwork" section of your profile

**Comment by iennonno** [comment]
Not really sure why there are people stating the issue is non-Steam games. It’s not. I have screenshots from the early 10s of World of Warcraft and more. This is a valid issue.

**Comment by Variapolis** [comment]
This would be nice to have fixed considering a large portion of people who use SteamOS are SteamDeck users that are emulating games which are going to be non-steam.

And as was stated previously. Uploading non-steam games works everywhere except in the actual Deck Game UI mode.

**Comment by safijari** [comment]
The reason for this is that the appID for nonsteam games is being truncated when a screenshot is saved. It's supposed to be a 32 bit unsigned integer but for some reason the last few bits get stripped before the screenshot is saved. Because of this Steam has no way of knowing which app the screenshots belong to. This is why they don't show up under the "your stuff" section, and why you can't upload them (the Steam client tries to match the folder of the screenshot with an appid and cannot do so).

Source: I'm the author of the Shotty plugin and had to navigate this mess to get the correct app names when aggregating screenshots.
